src/dimension_slice.c

```diff
@@ -68,21 +68,14 @@ ts_dimension_slice_create(int dimension_id, int64 range_start, int64 range_end)
 int
 ts_dimension_slice_cmp(const DimensionSlice *left, const DimensionSlice *right)
 {
-	if (left->fd.range_start == right->fd.range_start)
-	{
-		if (left->fd.range_end == right->fd.range_end)
-			return 0;
-
-		if (left->fd.range_end > right->fd.range_end)
-			return 1;
-
-		return -1;
-	}
+	int res = (left->fd.range_start > right->fd.range_start) -
```

Comment:
Even though this results in less instructions not sure this makes the code more readable

Reply:
I think it this is an easy to remember pattern to standardize on that is always "correct." Also, not sure it is that hard to understand; it basically just says that either (1) left  is greater than right, or (2) it is smaller, or (3) both are false = equal (0 - 0). Can aid the reader with a comment if that helps.

Reply:
Added some macros and comments to help with this.